src/sched/lthread.c

```diff
@@ -566,6 +566,9 @@ int lthread_create_primitive(
     lt->itls = tls;
     // for cloned threads, tls argument is a ptr to the Thread Control Block
     lt->tp = tls;
+    // set self pointer for fs segment
+    struct lthread_tcb_base *tcb = (struct lthread_tcb_base *)lt->tp;
+    tcb->self = lt->tp;
```

Comment:
Musl should be doing this for pthreads, is this just needed for the clone test? If so, we should probably fix that test (I think it’s actually fine to remove it and replace it with a pthread test that does the same thing, it’s only needed because pthreads weren’t using clone when I wrote it and we needed to test clone.).

Reply:
`tests/basic/pthread_tls` probably covers the behaviors the clone test covers - userspace thread creation and waiting on futex. It can't cover the deferred stack unmapping test though.

Reply:
This was added to support the clone test, I'll move this logic there. Should we add an assert here to check if the self pointer has been set?